front/src/common/api/osrdEditoastApi.ts

```diff
@@ -1984,6 +1985,17 @@ export type PathfindingInput = {
   ending: PathfindingTrackLocationInput;
   starting: PathfindingTrackLocationInput;
 };
+export type SwitchPortConnection = {
+  dst: string;
+  src: string;
+};
+export type SwitchType = {
+  groups: {
+    [key: string]: SwitchPortConnection[];
+  };
+  id: string;
```

Comment:
In the editor, we have a type for the switch possible ids : 
```
type TrackNodeTypeId =
  | 'crossing'
  | 'single_slip_switch'
  | 'double_slip_switch'
  | 'link'
  | 'point_switch';
```
Are you checking somewhere in the back that the front sends you one of these ids ? If not, should you have an enum for that ?